torch/csrc/utils/python_arg_parser.h

```diff
@@ -147,10 +147,10 @@ inline at::Scalar PythonArgs::scalar(int i) {
 
 inline at::Scalar PythonArgs::scalarWithDefault(int i, at::Scalar default_scalar) {
   if (!args[i]) return default_scalar;
-  if (PyFloat_Check(args[i])) {
-    return at::Scalar(THPUtils_unpackDouble(args[i]));
+  if (PyLong_CheckExact(args[i])) {
+    return at::Scalar(static_cast<int64_t>(THPUtils_unpackLong(args[i])));
```

Comment:
I've added a comment. This doesn't currently handle NumPy types (except np.float64, which is a subclass of `PyFloat`). There's a good chance we'll want to add support for that in the future.

The type-conversion is a little easier in `tensor.new(...)` so that supports NumPy types.